lib/create_azure_oauth_config.ts

```diff
@@ -0,0 +1,102 @@
+// Copyright 2023-2024 the Deno authors. All rights reserved. MIT license.
+import type { OAuth2ClientConfig } from "../deps.ts";
+import { getRequiredEnv } from "./get_required_env.ts";
+
+/**
+ * Returns the OAuth configuration for Azure.
+ *
+ * Requires `--allow-env[=AZURE_CLIENT_ID,AZURE_CLIENT_SECRET,AZURE_CLOUD_INSTANCE,AZURE_TENANT_ID]`
+ * permissions and environment variables:
+ * 1. `AZURE_CLIENT_ID`
+ * 2. `AZURE_CLIENT_SECRET`
+ * 3. `AZURE_CLOUD_INSTANCE`
+ * 4. `AZURE_TENANT_ID`
+ *
+ * @example
+ * ```ts
+ * import { createAzureOAuthConfig } from "https://deno.land/x/deno_kv_oauth@$VERSION/mod.ts";
+ *
+ * const oauthConfig = createAzureOAuthConfig({
+ *   envPrefix: "AZURE",
+ *   redirectUri: "http://localhost:8000/callback",
+ *   scope: "openid",
+ * });
+ * ```
+ *
+ * You can use a separate environment variable prefix with the envPrefix configureation.
+ * Requires `--allow-env[=MY_PREFIX_CLIENT_ID,MY_PREFIX_CLIENT_SECRET,MY_PREFIX_CLOUD_INSTANCE,MY_PREFIX_TENANT_ID]`
+ * permissions and environment variables:
+ * 1. `MY_PREFIX_CLIENT_ID`
+ * 2. `MY_PREFIX_CLIENT_SECRET`
+ * 3. `MY_PREFIX_CLOUD_INSTANCE`
+ * 4. `MY_PREFIX_TENANT_ID`
+ *
+ * @example
+ * ```ts
+ * import { createAzureOAuthConfig } from "https://deno.land/x/deno_kv_oauth@$VERSION/mod.ts";
+ *
+ * const oauthConfig = createAzureOAuthConfig({
+ *   envPrefix: "MY_PREFIX",
+ *   redirectUri: "http://localhost:8000/callback",
+ *   scope: "openid",
+ * });
+ * ```
+ *
+ * This also supports the ability to connect with Azure AD B2C instances using the
+ * proper configurations for AD B2C instances
+ *
+ * @see {@link https://learn.microsoft.com/en-us/entra/identity-platform/v2-protocols-oidc}
+ */
+export function createAzureOAuthConfig(config: {
```

Comment:
This function differs in its logic from other similar functions within this library. For ease of management of use, could you please try following the implementation of other `createXOAuthConfig()` functions in the library?